src/core/linalg/src/dense/4C_linalg_utils_scalar_interpolation.cpp

```diff
@@ -0,0 +1,470 @@
+
+    interp_scalar[j] = interp_value(0, 0);  // store the interpolated value
+  }
+
+  return interp_scalar;
+}
+
+/*--------------------------------------------------------------------*
+ *--------------------------------------------------------------------*/
+template <unsigned int loc_dim, unsigned int poly_order, unsigned int num_cofeffcients>
+std::vector<double> Core::LinAlg::ScalarInterpolation::ScalarInterpolator<loc_dim, poly_order,
+    num_cofeffcients>::log_moving_least_square(const std::vector<std::vector<double>>& scalar_data,
+    const std::vector<Core::LinAlg::Matrix<loc_dim, 1>>& ref_locs,
+    const Core::LinAlg::Matrix<loc_dim, 1>& interp_loc, const std::vector<double>& weights)
+{
+  std::vector<double> interp_scalar(scalar_data[0].size(), 0.0);
+
+  std::vector<std::vector<double>> log_scalar_data(scalar_data.size());
+
+  for (size_t i = 0; i < scalar_data.size(); ++i)
+  {
+    log_scalar_data[i].resize(scalar_data[i].size());
+    for (size_t j = 0; j < scalar_data[i].size(); ++j)
+    {
+      if (scalar_data[i][j] <= 0.0)
+        FOUR_C_THROW("Logarithmic moving least squares requires positive scalar data values.");
+      log_scalar_data[i][j] = std::log(scalar_data[i][j]);
+    }
+  }
+
+  // Use the moving least square method on the logarithmic data
+  std::vector<double> log_interp_scalar =
+      moving_least_square(log_scalar_data, ref_locs, interp_loc, weights);
+
+  // Exponentiate the result to get back to the original scale
+  for (size_t j = 0; j < log_interp_scalar.size(); ++j)
+    interp_scalar[j] = std::exp(log_interp_scalar[j]);
+
+  // check the interpolated scalar values
+  for (size_t j = 0; j < interp_scalar.size(); ++j)
+    if (interp_scalar[j] <= 0.0)
+      FOUR_C_THROW("Logarithmic moving least squares resulted in non-positive scalar values.");
+
+  // Return the interpolated scalar values
+  return interp_scalar;
+}
+
+/*--------------------------------------------------------------------*
+ *--------------------------------------------------------------------*/
+template <unsigned int loc_dim, unsigned int poly_order, unsigned int num_cofeffcients>
+std::vector<double> Core::LinAlg::ScalarInterpolation::ScalarInterpolator<loc_dim, poly_order,
+    num_cofeffcients>::get_interpolated_scalar(const std::vector<std::vector<double>>& scalar_data,
+    const std::vector<Core::LinAlg::Matrix<loc_dim, 1>>& ref_locs,
+    const Core::LinAlg::Matrix<loc_dim, 1>& interp_loc)
+{
+  // Check for size consistency
+  if (scalar_data.empty()) FOUR_C_THROW("Scalar data vector is empty.");
+  if (ref_locs.empty()) FOUR_C_THROW("Reference locations vector is empty.");
+  if (scalar_data.size() != ref_locs.size())
+    FOUR_C_THROW("Size mismatch: scalar_data vs ref_locs.");
+  if (ref_locs.size() < 1)
+    FOUR_C_THROW("At least one reference location is required for interpolation.");
+
+  auto weights =
+      this->calculate_normalized_weights(ref_locs, interp_loc, weight_func_, interp_params_);
```

Comment:
Is there any reason for using `this` when calling the static member function? I am not quite sure. What do you think, @c-p-schmidt?

```suggestion
  auto weights =
      calculate_normalized_weights(ref_locs, interp_loc, weight_func_, interp_params_);
```

Reply:
Method will moved out of the class and this line changes to 
`Core::LinAlg::calculate_normalized_weights`